app/gui2/src/providers/widgetRegistry/editHandler.ts

```diff
@@ -106,6 +109,22 @@ export class WidgetEditHandler {
     return editHandler
   }
 
+  child(widgetId: string, input: WidgetInput, myInteraction: WidgetEditHooks): WidgetEditHandler {
+    const { getPortEdit } = useGetPortEdit(
+      input.portId,
+      input[ArgumentInfoKey]?.argId,
+      injectInteractionHandler(),
+      this.portId,
+    )
+    return new WidgetEditHandler(
+      widgetId as WidgetId,
+      myInteraction,
+      this,
+      getPortEdit,
+      input.portId,
+    )
+  }
+
```

Comment:
I think this should not be needed anymore, given my comment above; the children widget should have their parent in WidgetInput received in constructor,